src/features/metaform-slice.ts

```diff
@@ -1,16 +1,24 @@
 import { createSlice, PayloadAction } from "@reduxjs/toolkit";
 import type { RootState } from "app/store";
-import { MetaformVersion } from "generated/client";
+import { MetaformVersion, MetaformSection, MetaformField } from "generated/client";
 
 interface MetaformState {
   metaformVersion?: MetaformVersion;
+  metaformSection?: MetaformSection;
+  metaformField?: MetaformField;
+  metaformSectionIndex?: number;
+  metaformFieldIndex?: number;
```

Comment:
Why we are storing them in redux? couldn't we just pass then as props? e.g. sectionId={sectionIndes} and access the section just using the index

Reply:
If it's really so, please just store the index but not the actual section/field, since you have the form in the redux already. You can access them by `form.sections[sectionIndex]` and `form.sections[sectionIndex].fields[fieldIndex]`. And btw whose request is it?